### ISSUE TITLE
 Cannot parse path to part: cannot parse maxTimestamp from partName - month out of range

### ISSUE THREAD

**Issue by nvtkaszpir** [VictoriaMetrics/VictoriaMetrics#2998]
**Describe the bug**
container crashing (single instance victoria metrics as statefulset, tried 1.77.2, tried 1.80.0 same error)
```text
{"ts":"2022-08-18T19:07:27.060Z","level":"fatal","caller":"VictoriaMetrics/app/vmstorage/main.go:113","msg":"cannot open a storage at /storage with -retentionPeriod=10y: cannot open table at \"/storage/data\": cannot open partitions in the table \"/storage/data\": cannot open partition \"2022_08\": cannot open small parts from \"/storage/data/small/2022_08\": cannot open part \"/storage/data/small/2022_08/8414552_62371_20220816121345.913_365366500914103712.714_170BB169DD29DC5C\": cannot parse path to part: cannot parse maxTimestamp from partName \"8414552_62371_20220816121345.913_365366500914103712.714_170BB169DD29DC5C\": parsing time \"365366500914103712.714\": month out of range"}
```

**To Reproduce**
No idea. Went back from holidays and I see the container is in CrashLoopBackoff.

**Expected behavior**
Container would open storage as is without problems.

**Logs**
```text
{"ts":"2022-08-18T19:07:25.651Z","level":"info","caller":"VictoriaMetrics/lib/logger/flag.go:12","msg":"build version: victoria-metrics-20220808-173538-tags-v1.80.0-0-gad00f4aaa"}
{"ts":"2022-08-18T19:07:25.651Z","level":"info","caller":"VictoriaMetrics/lib/logger/flag.go:13","msg":"command-line flags"}
{"ts":"2022-08-18T19:07:25.651Z","level":"info","caller":"VictoriaMetrics/lib/logger/flag.go:20","msg":"  -envflag.enable=\"true\""}
{"ts":"2022-08-18T19:07:25.651Z","level":"info","caller":"VictoriaMetrics/lib/logger/flag.go:20","msg":"  -envflag.prefix=\"VM_\""}
{"ts":"2022-08-18T19:07:25.651Z","level":"info","caller":"VictoriaMetrics/lib/logger/flag.go:20","msg":"  -loggerFormat=\"json\""}
{"ts":"2022-08-18T19:07:25.651Z","level":"info","caller":"VictoriaMetrics/lib/logger/flag.go:20","msg":"  -promscrape.config=\"/scrapeconfig/scrape.yml\""}
{"ts":"2022-08-18T19:07:25.651Z","level":"info","caller":"VictoriaMetrics/lib/logger/flag.go:20","msg":"  -retentionPeriod=\"10y\""}
{"ts":"2022-08-18T19:07:25.651Z","level":"info","caller":"VictoriaMetrics/lib/logger/flag.go:20","msg":"  -storageDataPath=\"/storage\""}
{"ts":"2022-08-18T19:07:25.651Z","level":"info","caller":"VictoriaMetrics/app/victoria-metrics/main.go:54","msg":"starting VictoriaMetrics at \":8428\"..."}
{"ts":"2022-08-18T19:07:25.651Z","level":"info","caller":"VictoriaMetrics/app/vmstorage/main.go:108","msg":"opening storage at \"/storage\" with -retentionPeriod=10y"}
{"ts":"2022-08-18T19:07:25.651Z","level":"info","caller":"VictoriaMetrics/lib/memory/memory.go:42","msg":"limiting caches to 20006311526 bytes, leaving 13337541018 bytes to the OS according to -memory.allowedPercent=60"}
{"ts":"2022-08-18T19:07:25.651Z","level":"info","caller":"VictoriaMetrics/lib/storage/storage.go:1037","msg":"loading MetricName->TSID cache from \"/storage/cache/metricName_tsid\"..."}
{"ts":"2022-08-18T19:07:25.695Z","level":"info","caller":"VictoriaMetrics/lib/storage/storage.go:1042","msg":"loaded MetricName->TSID cache from \"/storage/cache/metricName_tsid\" in 0.044 seconds; entriesCount: 69308; sizeBytes: 34013184"}
{"ts":"2022-08-18T19:07:25.695Z","level":"info","caller":"VictoriaMetrics/lib/storage/storage.go:1037","msg":"loading MetricID->TSID cache from \"/storage/cache/metricID_tsid\"..."}
{"ts":"2022-08-18T19:07:25.699Z","level":"info","caller":"VictoriaMetrics/lib/storage/storage.go:1042","msg":"loaded MetricID->TSID cache from \"/storage/cache/metricID_tsid\" in 0.004 seconds; entriesCount: 0; sizeBytes: 0"}
{"ts":"2022-08-18T19:07:25.699Z","level":"info","caller":"VictoriaMetrics/lib/storage/storage.go:1037","msg":"loading MetricID->MetricName cache from \"/storage/cache/metricID_metricName\"..."}
{"ts":"2022-08-18T19:07:25.704Z","level":"info","caller":"VictoriaMetrics/lib/storage/storage.go:1042","msg":"loaded MetricID->MetricName cache from \"/storage/cache/metricID_metricName\" in 0.004 seconds; entriesCount: 0; sizeBytes: 0"}
{"ts":"2022-08-18T19:07:25.704Z","level":"info","caller":"VictoriaMetrics/lib/storage/storage.go:886","msg":"loading curr_hour_metric_ids from \"/storage/cache/curr_hour_metric_ids\"..."}
{"ts":"2022-08-18T19:07:25.705Z","level":"info","caller":"VictoriaMetrics/lib/storage/storage.go:908","msg":"discarding /storage/cache/curr_hour_metric_ids, since it contains outdated hour; got 460148; want 461347"}
{"ts":"2022-08-18T19:07:25.705Z","level":"info","caller":"VictoriaMetrics/lib/storage/storage.go:886","msg":"loading prev_hour_metric_ids from \"/storage/cache/prev_hour_metric_ids\"..."}
{"ts":"2022-08-18T19:07:25.705Z","level":"info","caller":"VictoriaMetrics/lib/storage/storage.go:908","msg":"discarding /storage/cache/prev_hour_metric_ids, since it contains outdated hour; got 460147; want 461346"}
{"ts":"2022-08-18T19:07:25.705Z","level":"info","caller":"VictoriaMetrics/lib/storage/storage.go:842","msg":"loading next_day_metric_ids from \"/storage/cache/next_day_metric_ids\"..."}
{"ts":"2022-08-18T19:07:25.707Z","level":"info","caller":"VictoriaMetrics/lib/storage/storage.go:862","msg":"discarding /storage/cache/next_day_metric_ids, since it contains data for stale date; got 19172; want 19222"}
{"ts":"2022-08-18T19:07:25.724Z","level":"info","caller":"VictoriaMetrics/lib/mergeset/table.go:260","msg":"opening table \"/storage/indexdb/16F13A961D8B6E1C\"..."}
{"ts":"2022-08-18T19:07:25.828Z","level":"info","caller":"VictoriaMetrics/lib/mergeset/table.go:296","msg":"table \"/storage/indexdb/16F13A961D8B6E1C\" has been opened in 0.104 seconds; partsCount: 12; blocksCount: 45550, itemsCount: 18658910; sizeBytes: 526240726"}
{"ts":"2022-08-18T19:07:25.835Z","level":"info","caller":"VictoriaMetrics/lib/mergeset/table.go:260","msg":"opening table \"/storage/indexdb/16F13A961D8B6E1B\"..."}
{"ts":"2022-08-18T19:07:25.958Z","level":"info","caller":"VictoriaMetrics/lib/mergeset/table.go:296","msg":"table \"/storage/indexdb/16F13A961D8B6E1B\" has been opened in 0.123 seconds; partsCount: 0; blocksCount: 0, itemsCount: 0; sizeBytes: 0"}
{"ts":"2022-08-18T19:07:26.808Z","level":"info","caller":"VictoriaMetrics/lib/storage/partition.go:1607","msg":"opened part \"/storage/data/small/2022_07/59_59_20220731235956.193_20220731235956.689_17024EAC92A03DC7\" in 0.003 seconds"}
{"ts":"2022-08-18T19:07:26.809Z","level":"info","caller":"VictoriaMetrics/lib/storage/partition.go:1607","msg":"opened part \"/storage/data/small/2022_07/571935331_93669_20220701000000.220_20220701221850.414_16FD8B7B8E2813C9\" in 0.000 seconds"}
{"ts":"2022-08-18T19:07:26.809Z","level":"info","caller":"VictoriaMetrics/lib/storage/partition.go:1607","msg":"opened part \"/storage/data/small/2022_07/440877684_74801_20220729073440.414_20220730232300.414_17024EAC929FF26C\" in 0.000 seconds"}
{"ts":"2022-08-18T19:07:26.809Z","level":"info","caller":"VictoriaMetrics/lib/storage/partition.go:1607","msg":"opened part \"/storage/data/small/2022_07/56973_56958_20220731235923.617_20220731235930.414_17024EAC92A03DBE\" in 0.000 seconds"}
{"ts":"2022-08-18T19:07:26.810Z","level":"info","caller":"VictoriaMetrics/lib/storage/partition.go:1607","msg":"opened part \"/storage/data/small/2022_07/513306875_87502_20220727120749.626_20220728054350.417_17024EAC929F2967\" in 0.000 seconds"}
{"ts":"2022-08-18T19:07:26.833Z","level":"info","caller":"VictoriaMetrics/lib/storage/partition.go:1607","msg":"opened part \"/storage/data/small/2022_07/1000_1000_20220731235525.917_20220731235528.361_17024EAC92A03D8B\" in 0.023 seconds"}
{"ts":"2022-08-18T19:07:26.834Z","level":"info","caller":"VictoriaMetrics/lib/storage/partition.go:1607","msg":"opened part \"/storage/data/small/2022_07/4835848676_671405_20220701215649.615_20220708173816.092_16FD8B7B8E2A05F2\" in 0.001 seconds"}
{"ts":"2022-08-18T19:07:26.834Z","level":"info","caller":"VictoriaMetrics/lib/storage/partition.go:1607","msg":"opened part \"/storage/data/small/2022_07/423817_71737_20220731235708.121_20220731235820.414_17024EAC92A03DAE\" in 0.000 seconds"}
{"ts":"2022-08-18T19:07:26.835Z","level":"info","caller":"VictoriaMetrics/lib/storage/partition.go:1607","msg":"opened part \"/storage/data/small/2022_07/46_46_20220731235956.193_20220731235956.689_17024EAC92A03DC8\" in 0.000 seconds"}
{"ts":"2022-08-18T19:07:26.835Z","level":"info","caller":"VictoriaMetrics/lib/storage/partition.go:1607","msg":"opened part \"/storage/data/small/2022_07/14261_14246_20220731235937.357_20220731235955.818_17024EAC92A03DC3\" in 0.000 seconds"}
{"ts":"2022-08-18T19:07:26.836Z","level":"info","caller":"VictoriaMetrics/lib/storage/partition.go:1607","msg":"opened part \"/storage/data/small/2022_07/14261_14246_20220731235909.385_20220731235925.818_17024EAC92A03DBD\" in 0.000 seconds"}
{"ts":"2022-08-18T19:07:26.836Z","level":"info","caller":"VictoriaMetrics/lib/storage/partition.go:1607","msg":"opened part \"/storage/data/small/2022_07/56473_56458_20220731235919.044_20220731235940.414_17024EAC92A03DC0\" in 0.000 seconds"}
{"ts":"2022-08-18T19:07:26.837Z","level":"info","caller":"VictoriaMetrics/lib/storage/partition.go:1607","msg":"opened part \"/storage/data/small/2022_07/287495917_73750_20220729210530.414_20220730110520.414_17024EAC929FCCD5\" in 0.001 seconds"}
{"ts":"2022-08-18T19:07:26.837Z","level":"info","caller":"VictoriaMetrics/lib/storage/partition.go:1607","msg":"opened part \"/storage/data/small/2022_07/46404496_73282_20220731204538.617_20220731232246.757_17024EAC92A03BDD\" in 0.000 seconds"}
{"ts":"2022-08-18T19:07:26.838Z","level":"info","caller":"VictoriaMetrics/lib/storage/partition.go:1607","msg":"opened part \"/storage/data/small/2022_07/26006_25996_20220731235937.357_20220731235959.878_17024EAC92A03DC4\" in 0.000 seconds"}
{"ts":"2022-08-18T19:07:26.838Z","level":"info","caller":"VictoriaMetrics/lib/storage/partition.go:1607","msg":"opened part \"/storage/data/small/2022_07/555_555_20220731235947.626_20220731235959.626_17024EAC92A03DC5\" in 0.000 seconds"}
{"ts":"2022-08-18T19:07:26.839Z","level":"info","caller":"VictoriaMetrics/lib/storage/partition.go:1607","msg":"opened part \"/storage/data/small/2022_07/4542956755_598507_20220715113659.626_20240918210457.150_17024EAC929D549A\" in 0.001 seconds"}
{"ts":"2022-08-18T19:07:26.839Z","level":"info","caller":"VictoriaMetrics/lib/storage/partition.go:1607","msg":"opened part \"/storage/data/small/2022_07/500_500_20220731235715.917_20220731235718.248_17024EAC92A03DA2\" in 0.000 seconds"}
{"ts":"2022-08-18T19:07:26.840Z","level":"info","caller":"VictoriaMetrics/lib/storage/partition.go:1607","msg":"opened part \"/storage/data/small/2022_07/2218144_72777_20220731233338.617_20220731235130.414_17024EAC92A03D56\" in 0.001 seconds"}
{"ts":"2022-08-18T19:07:26.841Z","level":"info","caller":"VictoriaMetrics/lib/storage/partition.go:1607","msg":"opened part \"/storage/data/small/2022_07/1500_1495_20220731234925.917_20220731234929.878_17024EAC92A03D3D\" in 0.001 seconds"}
{"ts":"2022-08-18T19:07:26.841Z","level":"info","caller":"VictoriaMetrics/lib/storage/partition.go:1607","msg":"opened part \"/storage/data/small/2022_07/331688352_75691_20220729055309.626_20220729212020.414_17024EAC929FA2C3\" in 0.000 seconds"}
{"ts":"2022-08-18T19:07:26.842Z","level":"info","caller":"VictoriaMetrics/lib/storage/partition.go:1607","msg":"opened part \"/storage/data/small/2022_07/2082100_72753_20220731235044.247_20220731235620.414_17024EAC92A03D95\" in 0.001 seconds"}
{"ts":"2022-08-18T19:07:26.842Z","level":"info","caller":"VictoriaMetrics/lib/storage/partition.go:1607","msg":"opened part \"/storage/data/small/2022_07/105468713_98096_20220728163619.626_20220728210130.414_17024EAC929F5853\" in 0.000 seconds"}
{"ts":"2022-08-18T19:07:26.843Z","level":"info","caller":"VictoriaMetrics/lib/storage/partition.go:1607","msg":"opened part \"/storage/data/small/2022_07/15261_15246_20220731235929.197_20220731235945.818_17024EAC92A03DC1\" in 0.000 seconds"}
{"ts":"2022-08-18T19:07:26.843Z","level":"info","caller":"VictoriaMetrics/lib/storage/partition.go:1607","msg":"opened part \"/storage/data/small/2022_07/500_500_20220731235915.917_20220731235918.248_17024EAC92A03DBC\" in 0.000 seconds"}
{"ts":"2022-08-18T19:07:26.843Z","level":"info","caller":"VictoriaMetrics/lib/storage/partition.go:1607","msg":"opened part \"/storage/data/small/2022_07/13761_13746_20220731235919.044_20220731235935.818_17024EAC92A03DBF\" in 0.000 seconds"}
{"ts":"2022-08-18T19:07:26.861Z","level":"info","caller":"VictoriaMetrics/lib/storage/partition.go:1607","msg":"opened part \"/storage/data/small/2022_07/63106054_73264_20220731182925.917_20220731210420.414_17024EAC92A034D1\" in 0.017 seconds"}
{"ts":"2022-08-18T19:07:26.861Z","level":"info","caller":"VictoriaMetrics/lib/storage/partition.go:1607","msg":"opened part \"/storage/data/small/2022_07/4812951980_676820_20220708162400.220_20220715114220.414_16FD8B7B8E2BF483\" in 0.001 seconds"}
{"ts":"2022-08-18T19:07:26.862Z","level":"info","caller":"VictoriaMetrics/lib/storage/partition.go:1607","msg":"opened part \"/storage/data/small/2022_07/1000_1000_20220731235325.917_20220731235328.248_17024EAC92A03D71\" in 0.000 seconds"}
{"ts":"2022-08-18T19:07:26.862Z","level":"info","caller":"VictoriaMetrics/lib/storage/partition.go:1607","msg":"opened part \"/storage/data/small/2022_07/1500_1495_20220731235425.917_20220731235429.878_17024EAC92A03D7E\" in 0.000 seconds"}
{"ts":"2022-08-18T19:07:26.862Z","level":"info","caller":"VictoriaMetrics/lib/storage/partition.go:1607","msg":"opened part \"/storage/data/small/2022_07/567897014_98256_20220728053709.626_20220729055416.757_17024EAC929F7391\" in 0.000 seconds"}
{"ts":"2022-08-18T19:07:26.863Z","level":"info","caller":"VictoriaMetrics/lib/storage/partition.go:1607","msg":"opened part \"/storage/data/small/2022_07/1500_1495_20220731235225.917_20220731235229.878_17024EAC92A03D64\" in 0.000 seconds"}
{"ts":"2022-08-18T19:07:26.863Z","level":"info","caller":"VictoriaMetrics/lib/storage/partition.go:1607","msg":"opened part \"/storage/data/small/2022_07/53920644_73309_20220731155715.917_20220731183610.414_17024EAC92A02D3F\" in 0.000 seconds"}
{"ts":"2022-08-18T19:07:26.864Z","level":"info","caller":"VictoriaMetrics/lib/storage/partition.go:1607","msg":"opened part \"/storage/data/small/2022_07/40005335_73258_20220731143159.626_20220731162940.414_17024EAC92A026D0\" in 0.000 seconds"}
{"ts":"2022-08-18T19:07:26.864Z","level":"info","caller":"VictoriaMetrics/lib/storage/partition.go:1607","msg":"opened part \"/storage/data/small/2022_07/424696_71618_20220731235609.000_20220731235720.414_17024EAC92A03DA1\" in 0.000 seconds"}
{"ts":"2022-08-18T19:07:26.864Z","level":"info","caller":"VictoriaMetrics/lib/storage/partition.go:1607","msg":"opened part \"/storage/data/small/2022_07/78_78_20220731235947.626_20220731235958.361_17024EAC92A03DC6\" in 0.000 seconds"}
{"ts":"2022-08-18T19:07:26.865Z","level":"info","caller":"VictoriaMetrics/lib/storage/partition.go:1607","msg":"opened part \"/storage/data/small/2022_07/500_500_20220731235815.917_20220731235818.248_17024EAC92A03DAF\" in 0.000 seconds"}
{"ts":"2022-08-18T19:07:26.865Z","level":"info","caller":"VictoriaMetrics/lib/storage/partition.go:1607","msg":"opened part \"/storage/data/small/2022_07/55973_55958_20220731235929.197_20220731235950.414_17024EAC92A03DC2\" in 0.000 seconds"}
{"ts":"2022-08-18T19:07:26.865Z","level":"info","caller":"VictoriaMetrics/lib/storage/partition.go:1607","msg":"opened part \"/storage/data/small/2022_07/4026797598_507101_20220721205250.418_20220727123046.756_17024EAC929EF483\" in 0.000 seconds"}
{"ts":"2022-08-18T19:07:26.866Z","level":"info","caller":"VictoriaMetrics/lib/storage/partition.go:1607","msg":"opened part \"/storage/data/small/2022_07/63557426_73398_20220730232555.917_20220731032520.414_17024EAC929FFEC9\" in 0.000 seconds"}
{"ts":"2022-08-18T19:07:26.866Z","level":"info","caller":"VictoriaMetrics/lib/storage/partition.go:1607","msg":"opened part \"/storage/data/small/2022_07/10055433_72896_20220731231559.626_20220731234620.414_17024EAC92A03D13\" in 0.000 seconds"}
{"ts":"2022-08-18T19:07:26.867Z","level":"info","caller":"VictoriaMetrics/lib/storage/partition.go:1607","msg":"opened part \"/storage/data/small/2022_07/333710571_74032_20220730231419.615_20220731145534.824_17024EAC92A02206\" in 0.000 seconds"}
{"ts":"2022-08-18T19:07:26.867Z","level":"info","caller":"VictoriaMetrics/lib/storage/partition.go:1607","msg":"opened part \"/storage/data/small/2022_07/7998_1993_20220731234325.917_20220731235131.581_17024EAC92A03D57\" in 0.000 seconds"}
{"ts":"2022-08-18T19:07:26.867Z","level":"info","caller":"VictoriaMetrics/lib/storage/partition.go:1607","msg":"opened part \"/storage/data/small/2022_07/12614150_72960_20220731211515.917_20220731221956.757_17024EAC92A038A7\" in 0.000 seconds"}
{"ts":"2022-08-18T19:07:26.868Z","level":"info","caller":"VictoriaMetrics/lib/storage/partition.go:1607","msg":"opened part \"/storage/data/small/2022_07/427402_72579_20220731235809.239_20220731235920.414_17024EAC92A03DBB\" in 0.000 seconds"}
{"ts":"2022-08-18T19:07:27.054Z","level":"info","caller":"VictoriaMetrics/lib/storage/partition.go:1607","msg":"opened part \"/storage/data/small/2022_08/3416552680_433071_20220810065109.626_20220815205820.414_17070F792E8E41AD\" in 0.001 seconds"}
{"ts":"2022-08-18T19:07:27.055Z","level":"info","caller":"VictoriaMetrics/lib/storage/partition.go:1607","msg":"opened part \"/storage/data/small/2022_08/1982_1982_20220816124530.834_20220816124534.098_170BB169DD29DC54\" in 0.000 seconds"}
{"ts":"2022-08-18T19:07:27.055Z","level":"info","caller":"VictoriaMetrics/lib/storage/partition.go:1607","msg":"opened part \"/storage/data/small/2022_08/47270237_88026_20220816001759.626_20220816021825.913_170BB169DD29BB8D\" in 0.000 seconds"}
{"ts":"2022-08-18T19:07:27.056Z","level":"info","caller":"VictoriaMetrics/lib/storage/partition.go:1607","msg":"opened part \"/storage/data/small/2022_08/3152134150_431275_20220803021908.617_20220810070810.414_17070F792E8CA7FE\" in 0.000 seconds"}
{"ts":"2022-08-18T19:07:27.056Z","level":"info","caller":"VictoriaMetrics/lib/storage/partition.go:1607","msg":"opened part \"/storage/data/small/2022_08/40282311_62659_20220816102750.834_20220816122300.704_170BB169DD29DB23\" in 0.000 seconds"}
{"ts":"2022-08-18T19:07:27.056Z","level":"info","caller":"VictoriaMetrics/lib/storage/partition.go:617","msg":"waiting for stale parts remover to stop on \"/storage/data/small/2022_07\"..."}
{"ts":"2022-08-18T19:07:27.056Z","level":"info","caller":"VictoriaMetrics/lib/storage/partition.go:620","msg":"stale parts remover stopped in 0.000 seconds on \"/storage/data/small/2022_07\""}
{"ts":"2022-08-18T19:07:27.057Z","level":"info","caller":"VictoriaMetrics/lib/storage/partition.go:622","msg":"waiting for inmemory parts flusher to stop on \"/storage/data/small/2022_07\"..."}
{"ts":"2022-08-18T19:07:27.057Z","level":"info","caller":"VictoriaMetrics/lib/storage/partition.go:625","msg":"inmemory parts flusher stopped in 0.000 seconds on \"/storage/data/small/2022_07\""}
{"ts":"2022-08-18T19:07:27.057Z","level":"info","caller":"VictoriaMetrics/lib/storage/partition.go:627","msg":"waiting for raw rows flusher to stop on \"/storage/data/small/2022_07\"..."}
{"ts":"2022-08-18T19:07:27.057Z","level":"info","caller":"VictoriaMetrics/lib/storage/partition.go:630","msg":"raw rows flusher stopped in 0.000 seconds on \"/storage/data/small/2022_07\""}
{"ts":"2022-08-18T19:07:27.057Z","level":"info","caller":"VictoriaMetrics/lib/storage/partition.go:632","msg":"waiting for small part mergers to stop on \"/storage/data/small/2022_07\"..."}
{"ts":"2022-08-18T19:07:27.057Z","level":"info","caller":"VictoriaMetrics/lib/storage/partition.go:635","msg":"small part mergers stopped in 0.000 seconds on \"/storage/data/small/2022_07\""}
{"ts":"2022-08-18T19:07:27.057Z","level":"info","caller":"VictoriaMetrics/lib/storage/partition.go:637","msg":"waiting for big part mergers to stop on \"/storage/data/big/2022_07\"..."}
{"ts":"2022-08-18T19:07:27.057Z","level":"info","caller":"VictoriaMetrics/lib/storage/partition.go:640","msg":"big part mergers stopped in 0.000 seconds on \"/storage/data/big/2022_07\""}
{"ts":"2022-08-18T19:07:27.057Z","level":"info","caller":"VictoriaMetrics/lib/storage/partition.go:642","msg":"flushing inmemory parts to files on \"/storage/data/small/2022_07\"..."}
{"ts":"2022-08-18T19:07:27.057Z","level":"info","caller":"VictoriaMetrics/lib/storage/partition.go:666","msg":"0 inmemory parts have been flushed to files in 0.000 seconds on \"/storage/data/small/2022_07\""}
{"ts":"2022-08-18T19:07:27.058Z","level":"info","caller":"VictoriaMetrics/lib/mergeset/table.go:333","msg":"waiting for raw items flusher to stop on \"/storage/indexdb/16F13A961D8B6E1C\"..."}
{"ts":"2022-08-18T19:07:27.058Z","level":"info","caller":"VictoriaMetrics/lib/mergeset/table.go:336","msg":"raw items flusher stopped in 0.000 seconds on \"/storage/indexdb/16F13A961D8B6E1C\""}
{"ts":"2022-08-18T19:07:27.058Z","level":"info","caller":"VictoriaMetrics/lib/mergeset/table.go:338","msg":"waiting for converters to stop on \"/storage/indexdb/16F13A961D8B6E1C\"..."}
{"ts":"2022-08-18T19:07:27.058Z","level":"info","caller":"VictoriaMetrics/lib/mergeset/table.go:341","msg":"converters stopped in 0.000 seconds on \"/storage/indexdb/16F13A961D8B6E1C\""}
{"ts":"2022-08-18T19:07:27.058Z","level":"info","caller":"VictoriaMetrics/lib/mergeset/table.go:343","msg":"waiting for part mergers to stop on \"/storage/indexdb/16F13A961D8B6E1C\"..."}
{"ts":"2022-08-18T19:07:27.058Z","level":"info","caller":"VictoriaMetrics/lib/mergeset/table.go:346","msg":"part mergers stopped in 0.000 seconds on \"/storage/indexdb/16F13A961D8B6E1C\""}
{"ts":"2022-08-18T19:07:27.058Z","level":"info","caller":"VictoriaMetrics/lib/mergeset/table.go:348","msg":"flushing inmemory parts to files on \"/storage/indexdb/16F13A961D8B6E1C\"..."}
{"ts":"2022-08-18T19:07:27.058Z","level":"info","caller":"VictoriaMetrics/lib/mergeset/table.go:372","msg":"0 inmemory parts have been flushed to files in 0.000 seconds on \"/storage/indexdb/16F13A961D8B6E1C\""}
{"ts":"2022-08-18T19:07:27.058Z","level":"info","caller":"VictoriaMetrics/lib/mergeset/table.go:374","msg":"waiting for flush callback worker to stop on \"/storage/indexdb/16F13A961D8B6E1C\"..."}
{"ts":"2022-08-18T19:07:27.058Z","level":"info","caller":"VictoriaMetrics/lib/mergeset/table.go:377","msg":"flush callback worker stopped in 0.000 seconds on \"/storage/indexdb/16F13A961D8B6E1C\""}
{"ts":"2022-08-18T19:07:27.059Z","level":"info","caller":"VictoriaMetrics/lib/mergeset/table.go:333","msg":"waiting for raw items flusher to stop on \"/storage/indexdb/16F13A961D8B6E1B\"..."}
{"ts":"2022-08-18T19:07:27.059Z","level":"info","caller":"VictoriaMetrics/lib/mergeset/table.go:336","msg":"raw items flusher stopped in 0.000 seconds on \"/storage/indexdb/16F13A961D8B6E1B\""}
{"ts":"2022-08-18T19:07:27.059Z","level":"info","caller":"VictoriaMetrics/lib/mergeset/table.go:338","msg":"waiting for converters to stop on \"/storage/indexdb/16F13A961D8B6E1B\"..."}
{"ts":"2022-08-18T19:07:27.059Z","level":"info","caller":"VictoriaMetrics/lib/mergeset/table.go:341","msg":"converters stopped in 0.000 seconds on \"/storage/indexdb/16F13A961D8B6E1B\""}
{"ts":"2022-08-18T19:07:27.059Z","level":"info","caller":"VictoriaMetrics/lib/mergeset/table.go:343","msg":"waiting for part mergers to stop on \"/storage/indexdb/16F13A961D8B6E1B\"..."}
{"ts":"2022-08-18T19:07:27.059Z","level":"info","caller":"VictoriaMetrics/lib/mergeset/table.go:346","msg":"part mergers stopped in 0.000 seconds on \"/storage/indexdb/16F13A961D8B6E1B\""}
{"ts":"2022-08-18T19:07:27.059Z","level":"info","caller":"VictoriaMetrics/lib/mergeset/table.go:348","msg":"flushing inmemory parts to files on \"/storage/indexdb/16F13A961D8B6E1B\"..."}
{"ts":"2022-08-18T19:07:27.059Z","level":"info","caller":"VictoriaMetrics/lib/mergeset/table.go:372","msg":"0 inmemory parts have been flushed to files in 0.000 seconds on \"/storage/indexdb/16F13A961D8B6E1B\""}
{"ts":"2022-08-18T19:07:27.059Z","level":"info","caller":"VictoriaMetrics/lib/mergeset/table.go:374","msg":"waiting for flush callback worker to stop on \"/storage/indexdb/16F13A961D8B6E1B\"..."}
{"ts":"2022-08-18T19:07:27.059Z","level":"info","caller":"VictoriaMetrics/lib/mergeset/table.go:377","msg":"flush callback worker stopped in 0.000 seconds on \"/storage/indexdb/16F13A961D8B6E1B\""}
{"ts":"2022-08-18T19:07:27.060Z","level":"fatal","caller":"VictoriaMetrics/app/vmstorage/main.go:113","msg":"cannot open a storage at /storage with -retentionPeriod=10y: cannot open table at \"/storage/data\": cannot open partitions in the table \"/storage/data\": cannot open partition \"2022_08\": cannot open small parts from \"/storage/data/small/2022_08\": cannot open part \"/storage/data/small/2022_08/8414552_62371_20220816121345.913_365366500914103712.714_170BB169DD29DC5C\": cannot parse path to part: cannot parse maxTimestamp from partName \"8414552_62371_20220816121345.913_365366500914103712.714_170BB169DD29DC5C\": parsing time \"365366500914103712.714\": month out of range"}
```

**Screenshots**
Nothing here.

**Version**
- 1.77.2
- pgrade to 1.80.0 - same error

**Used command-line flags**
```text

--retentionPeriod=10y --storageDataPath=/storage -promscrape.config=/scrapeconfig/scrape.yml --envflag.enable=true --envflag.prefix=VM_ --loggerFormat=json

```

**Comment by nvtkaszpir** [comment]
```text
21:55:40 root@bagno /var/lib/rancher/k3s/storage/pvc-8928ac37-3162-4604-ae7a-e5e86f1b00de_victoria-metrics_server-volume-vms-victoria-metrics-single-server-0/data/small/2022_08 # ls -la
total 56
drwxr-xr-x 14 root root 4096 Aug 18 21:53 .
drwxr-xr-x  7 root root 4096 Aug  1 02:00 ..
drwxr-xr-x  2 root root 4096 Aug 16 14:45 1982_1982_20220816124530.834_20220816124534.098_170BB169DD29DC54
drwxr-xr-x  2 root root 4096 Aug  5 22:44 2555336161_358887_20220801000000.085_20220805204310.414_17070F792E8B628F
drwxr-xr-x  2 root root 4096 Aug 16 12:32 277893762_90138_20220815190930.418_20220816103155.913_170BB169DD29D557
drwxr-xr-x  2 root root 4096 Aug 10 09:09 3152134150_431275_20220803021908.617_20220810070810.414_17070F792E8CA7FE
drwxr-xr-x  2 root root 4096 Aug 15 23:01 3416552680_433071_20220810065109.626_20220815205820.414_17070F792E8E41AD
drwxr-xr-x  2 root root 4096 Aug 16 14:23 40282311_62659_20220816102750.834_20220816122300.704_170BB169DD29DB23
drwxr-xr-x  2 root root 4096 Aug 16 04:18 47270237_88026_20220816001759.626_20220816021825.913_170BB169DD29BB8D
drwxr-xr-x  2 root root 4096 Aug 16 14:46 8414552_62371_20220816121345.913_365366500914103712.714_170BB169DD29DC5C
drwxr-xr-x  2 root root 4096 Aug 16 14:45 929_929_20220816124535.557_20220816124537.600_170BB169DD29DC55
drwxr-xr-x  2 root root 4096 Aug 16 14:46 9688_9673_20220816124600.278_20220816124615.817_170BB169DD29DC5D
drwxr-xr-x  2 root root 4096 Aug 18 21:53 tmp
drwxr-xr-x  2 root root 4096 Aug 18 21:53 txn
```

**Comment by nvtkaszpir** [comment]
after removing physically directory `8414552_62371_20220816121345.913_365366500914103712.714_170BB169DD29DC5C` the post spawned succesfully.

**Comment by nvtkaszpir** [comment]
[Uploading offending-data.tar.gz…]()
The data chunk that was causing the issues.

**Comment by dmitryk-dk** [comment]
Hi @nvtkaszpir ! We have investigated the issue. One possible explanation of the bug - memory corruption or problem in the physical storage. 
Can you share some information about where you run your cluster?

**Comment by nvtkaszpir** [comment]
single node on non-ecc memory host with k3s v1.22.7+k3s1 with zfs storage as main layer. as raidz2
at 24th Aug zfs scrub repaired 240k of data with 0 errors (only 3 checksums errors)

**Comment by dmitryk-dk** [comment]
Hi @nvtkaszpir ! We have double checked our code and we didn't found any problem with it. 
So we confirmed memory corruption. Can we close the issue?

**Comment by nvtkaszpir** [comment]
Any option to somehow restore that data by for example naming properly directory and testing if VM could reprocess it?
I asked on slack few days ago but no response in that matter.

**Comment by dmitryk-dk** [comment]
I will check this and tell you more. Need some time. Sorry for delay, maybe we missed you question in slack

**Comment by dmitryk-dk** [comment]
Hi @nvtkaszpir ! Sorry for delay. At that moment we haven't any app which can restore corrupted data, but you can implement some small script which can read each part of the data and check timestamps and correctness of the block and add it to the folder where indexes saved. It is one possible solution to restore data. 
We have plans to implement this solution but not now.

**Comment by valyala** [comment]
@nvtkaszpir , the commit 5fa952549893d90ef9d1074d4d61ab4134e7a010 should detect similar on-disk data corruption at earlier stages during [background merge of the parts](https://docs.victoriametrics.com/#storage) with the corrupted timestamps. This should prevent from creating an output part with invalid timestamp in its name. This commit will be included in the next release of VitoriaMetrics.

See also https://github.com/VictoriaMetrics/VictoriaMetrics/issues/3011

**Comment by valyala** [comment]
FYI, the commit https://github.com/VictoriaMetrics/VictoriaMetrics/commit/5fa952549893d90ef9d1074d4d61ab4134e7a010 has been included in [VictoriaMetrics v1.82.0](https://github.com/VictoriaMetrics/VictoriaMetrics/releases).
